Enterprise-CMCS/macpro-mdct-mcr
# Number schema tweak

## Summary
Be a little more restrictive in what passes as a "number" in our `yup` schemas.

### Description
The only change here is a regex change, from "Must contain at least one digit, period, or comma" to "Starts with possibly a dash, then possibly a dot, then definitely a digit. Must not contain subsequent characters other than digits, periods, and commas."

This continues to allow values such as `"-.2"`, as well as obviously-incorrect values such as `"1,,,,,....2"`, but at least prevents values such as `"."`, `"2spooky4me"`, and `"Hello, world!"`. Essentially, we're upgrading from "Make sure it's a string" to "Make sure it looks kind of like a number". Stricter validation occurs later in the pipeline (I believe).

This change applies to validation on both the frontend and the backend.

### Related ticket
N/A

### How to test
This change is covered by existing unit tests; see the changed files for details. Execute the tests with `yarn test`, both in services/app-api and services/ui-src.

### Important updates
N/A

### Author checklist
